LeagueToolkit/IO/OBJ/OBJFile.cs

```diff
@@ -13,44 +14,147 @@ public class OBJFile
         public List<Vector3> Vertices { get; set; } = new List<Vector3>();
         public List<Vector2> UVs { get; set; } = new List<Vector2>();
         public List<Vector3> Normals { get; set; } = new List<Vector3>();
+        public List<OBJGroup> Groups { get; set; } = new List<OBJGroup>();
+        public string MaterialsFile { get; set; } = "";
+        public bool IsSmooth { get; set; } = false;
         public List<OBJFace> Faces { get; set; } = new List<OBJFace>();
 
         public OBJFile(List<Vector3> vertices, List<uint> indices)
         {
             this.Vertices = vertices;
-            for (int i = 0; i < indices.Count; i += 3)
-            {
-                uint[] faceIndices = new uint[] { indices[i], indices[i + 1], indices[i + 2] };
-                this.Faces.Add(new OBJFace(faceIndices));
-            }
+            this.Groups.Add(new OBJGroup("default", "defaultMat", indices));
         }
 
         public OBJFile(List<Vector3> vertices, List<uint> indices, List<Vector2> uvs)
         {
             this.Vertices = vertices;
             this.UVs = uvs;
-            for (int i = 0; i < indices.Count; i += 3)
-            {
-                uint[] faceIndices = new uint[] { indices[i], indices[i + 1], indices[i + 2] };
-                this.Faces.Add(new OBJFace(faceIndices, faceIndices));
-            }
+            this.Groups.Add(new OBJGroup("default", "defaultMat", indices, indices));
         }
 
         public OBJFile(List<Vector3> vertices, List<uint> indices, List<Vector2> uvs, List<Vector3> normals)
         {
             this.Vertices = vertices;
             this.UVs = uvs;
             this.Normals = normals;
-            for (int i = 0; i < indices.Count; i += 3)
+            this.Groups.Add(new OBJGroup("default", "defaultMat", indices, indices, indices));
+        }
+
+        public OBJFile(List<Vector3> vertices, Dictionary<Tuple<string, string>, List<uint>> groups, List<Vector2> uvs, List<Vector3> normals, string mtlFile = "")
```

Comment:
Is there anything you could do to simplify this `Dictionary<Tuple<string, string>, List<uint>>`? Perhaps using deconstruction with named variables at least, or just by passing an `OBJGroup` directly? Because I'm really struggling to understand what this construct is accomplishing.

Reply:
Resolved to `List<OBJGroup>`.